### PR TITLE
Lyche link-checking

### LINKED ISSUE #5: Set up linkchecking
CI infrastructure should mirror the linkchecker as configured in [OSCAL Pages](https://github.com/usnistgov/OSCAL-Pages)

### PR DESCRIPTION
**Pull request by nikitawootten-nist**
This PR sets up link checking with Hugo (#5)

## Progress

- [x] Lychee link checking wired into Makefile
- [x] Update `pages` workflow to optionally perform link checking (allow the result to be ignored on `workflow_dispatch`)
- [x] Fix makefile rebuilding Hugo on dependent targets
- [x] ~~Fix Lychee behavior on relative links~~ Fix the model documentation pipeline's relative link handling: Awaiting https://github.com/usnistgov/metaschema-xslt/pull/75
- [x] Fix all links
- [x] Update `pages` workflow to:
     - [x] run periodic checks
     - [x] produce PR comments
     - [x] upload results as an artifact
- [x] Document usage

## Side-effects

- Parametrized the pages GHA workflow for use with `workflow_dispatch` events
- Added aliases for pages that are no longer needed, and bad links from old OSCAL Metaschemas
- Fixed about 900 dead links

### CODE REVIEW

**Comment by nikitawootten-nist** [comment]
Note: the workflow will fail due to new permissions being added to the action. Refer to the test PR for results: https://github.com/nikitawootten-nist/OSCAL-Reference/pull/1

**Review by wendellpiez — APPROVED** [review]
Looks awesome, nice work on a pesky problem.

Interesting to think about: whether OSCAL repo wants its own XSpec, whether calling, or sometimes switching in for the XSpec lower down.

File: site/content/concepts/identifier-use/index.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en-us">
+
+<!--
+    Some older versions of OSCAL refer to the identifier use page with a relative path.
+
+    This alias ensures no links will break even for those older versions.
+-->
```
Comment (aj-stein-nist) [inline site/content/concepts/identifier-use/index.html:8]: NB: I am fine with this, shouldn't we just fix the aliases in OSCAL-Pages as well?

File: site/content/concepts/layer/assessment/assessment-plan/index.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html lang="en-us">
+
+<!--
+    Some older versions of OSCAL refer to the assessment plan concept page with a relative path.
+
+    This alias ensures no links will break even for those older versions.
+-->
```
Comment (aj-stein-nist) [inline site/content/concepts/layer/assessment/assessment-plan/index.html:8]: NB: Ditto as well re the above comment about another missing alias.

**Comment by nikitawootten-nist** [comment]
@iMichaela @aj-stein-nist should the release-notes be removed/aliased to GitHub's release page as part of this PR?

File: site/content/models/datatypes.md [hunk 3]
```diff
@@ -1,704 +0,0 @@
----
```
Comment (aj-stein-nist) [inline site/content/models/datatypes.md:1]: NB: Thank you for deleting the obsolete page now that it has been removed.

**Review by Compton-US — APPROVED** [review]
Verified that the ssp and cdef links are working.

**Comment by aj-stein-nist** [comment]
> @iMichaela @aj-stein-nist should the release-notes be removed/aliased to GitHub's release page as part of this PR?

I am fine with removing the redundant data or we need to update a checklist and start adding a page with PR review to lag a release. @iMichaela suggested as much: I think if we want to do that, we want to update the ("new") news page after the release is drafted. That means I am happy moving the currently (and frequently) out of sync release page (since we can edit GitHub notes quickly, the website) less so.

We can wait to see if Michaela agrees.

**Review by aj-stein-nist — APPROVED** [review]

File: site/content/concepts/identifier-use/index.html (hunk 1 above)
Comment (nikitawootten-nist) [reply site/content/concepts/identifier-use/index.html:8]: The aliases are fixed there too (via https://github.com/usnistgov/OSCAL-Pages/pull/51). This fix is referring to older source metaschemas that refer to pages with relative links (such as a remark directly to `/concepts/identifier-use/`, not a URL).

Comment (aj-stein-nist) [reply site/content/concepts/identifier-use/index.html:8]: That makes sense. I approved, but I will resolve this anyway.

**Comment by aj-stein-nist** [comment]
I had to understand [the CI/CD error](https://github.com/usnistgov/OSCAL-Reference/actions/runs/6315111512/job/17146917732?pr=22) but it appears to relate to the permissions because the PR opened from a fork, not a branch of the upstream, @nikitawootten-nist. I am still happy with it and it stays approved.

**Comment by nikitawootten-nist** [comment]
> I had to understand [the CI/CD error](https://github.com/usnistgov/OSCAL-Reference/actions/runs/6315111512/job/17146917732?pr=22) but it appears to relate to the permissions because the PR opened from a fork, not a branch of the upstream, @nikitawootten-nist. I am still happy with it and it stays approved.

Unfortunately GitHub's UX for PRs from a fork isn't great. Any new permissions are denied without explanation. I have a twin PR on my fork demonstrating that the commits do build and produce linkcheck comments: https://github.com/nikitawootten-nist/OSCAL-Reference/pull/1

**Comment by aj-stein-nist** [comment]
> We can wait to see if Michaela agrees.

@iMichaela agreed to this approach. @nikitawootten-nist will work on removing releases page, AJ will update procedures for release docs to update it and include an update to the news page.

**Comment by nikitawootten-nist** [comment]
Final checks passed: https://github.com/nikitawootten-nist/OSCAL-Reference/actions/runs/6329066790/job/17188603346

Merging now

**Comment by aj-stein-nist** [comment]
Guidance updated in core wiki in the commit below.

https://github.com/usnistgov/OSCAL/wiki/OSCAL-Release-Checklist/45da2cb9fd540ef9e1a8ed195940e4f5326913f6

News page approach backported per usnistgov/OSCAL-Pages#52.

**Pull request merged**